onnxruntime/core/session/custom_ops.cc

```diff
@@ -140,15 +140,15 @@ common::Status CreateCustomRegistry(const std::vector<OrtCustomOpDomain*>& op_do
       KernelDefBuilder def_builder;
       def_builder.SetName(op->GetName(op))
           .SetDomain(domain->domain_)
-          .SinceVersion(1);
+          .SinceVersion(1)
+          .TypeConstraint("T", DataTypeImpl::AllTensorTypes());
```

Comment:
Is this okay to add even if none of the inputs/outputs use ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED?

Reply:
No - developers are still required to return a fixed number of input/output types, among which if a returned type is ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED, we will give it a type name T which could be of any supported ort type.

Reply:
What's the objective of adding AllTensorTypes all the time?